File: app/src/main/java/com/example/tirtir_mcommerce/model/MarketingActivity.java
package com.example.tirtir_mcommerce.model;

import com.google.gson.annotations.SerializedName;

public class MarketingActivity {

    @SerializedName("_id")
    private String id;

    @SerializedName("type")
    private String type; // success, system, draft

    @SerializedName("title")
    private String title;

    @SerializedName("targetOrStatus")
    private String targetOrStatus;

    @SerializedName("createdAt")
    private String createdAt;

    public String getId() { return id; }
    public String getType() { return type; }
    public String getTitle() { return title; }
    public String getTargetOrStatus() { return targetOrStatus; }
    public String getCreatedAt() { return createdAt; }
}

File: scratch-backend/models/MarketingActivity.js
const mongoose = require('mongoose');

const marketingActivitySchema = new mongoose.Schema({
    type: { type: String, enum: ['success', 'system', 'draft'], required: true },
    title: { type: String, required: true },
    targetOrStatus: { type: String, required: true }
}, { timestamps: true });

module.exports = mongoose.models.MarketingActivity || mongoose.model('MarketingActivity', marketingActivitySchema);
